database/sql/src/main/scala/cromwell/database/slick/WorkflowStoreSlickDatabase.scala

```diff
@@ -35,22 +49,29 @@ trait WorkflowStoreSlickDatabase extends WorkflowStoreSqlDatabase {
     runTransaction(action) void
   }
 
-  override def queryWorkflowStoreEntries(limit: Int, queryWorkflowState: String, updateWorkflowState: String)
-                                        (implicit ec: ExecutionContext): Future[Seq[WorkflowStoreEntry]] = {
+  override def fetchRunnableWorkflows(limit: Int)
+                                     (implicit ec: ExecutionContext): Future[Seq[WorkflowStoreEntry]] = {
     val action = for {
-      workflowStoreEntries <- dataAccess.workflowStoreEntriesForWorkflowState((queryWorkflowState, limit.toLong)).result
-      _ <- DBIO.sequence(workflowStoreEntries map updateWorkflowStateForWorkflowExecutionUuid(updateWorkflowState))
+      workflowStoreEntries <- dataAccess.fetchStartableWorkflows(limit.toLong).result
+      _ <- DBIO.sequence(workflowStoreEntries map updateWorkflowStateAndRestartedForWorkflowExecutionUuid)
     } yield workflowStoreEntries
+
     runTransaction(action)
   }
 
-  private def updateWorkflowStateForWorkflowExecutionUuid(updateWorkflowState: String)
-                                                         (workflowStoreEntry: WorkflowStoreEntry)
+  private def updateWorkflowStateAndRestartedForWorkflowExecutionUuid(workflowStoreEntry: WorkflowStoreEntry)
                                                          (implicit ec: ExecutionContext): DBIO[Unit] = {
     val workflowExecutionUuid = workflowStoreEntry.workflowExecutionUuid
+    val updateState = workflowStoreEntry.workflowState match {
+        // Submitted workflows become running when fetched
+      case WorkflowStoreState.Submitted => WorkflowStoreState.Running
```

Comment:
This isn't saying that all submitted workflows go to running on restart, does it?

Reply:
No it's the state the workflow is updated to when it's fetched by the WMA. Submitted becomes Running, the rest (Running or Aborting with restarted = true) stay the same (but the restarted flag is set to false)